agent/exploits/cve_2024_6387.py

```diff
@@ -0,0 +1,95 @@
+import socket
+import re
+
+from agent import definitions, exploits_registry
+from ostorlab.agent.kb import kb
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+
+VULNERABILITY_TITLE = "OpenSSH Remote Code Execution Vulnerability"
+VULNERABILITY_REFERENCE = "CVE-2024-6387"
+VULNERABILITY_DESCRIPTION = (
+    "A remote code execution vulnerability in multiple versions of OpenSSH allows remote attackers to execute "
+    "arbitrary code, granting full root access. The vulnerability affects the default configuration and does not "
+    "require user interaction."
+)
+DEFAULT_TIMEOUT = 90
+
+
+def _create_vulnerability(
+    target: definitions.Target, details: str
+) -> definitions.Vulnerability:
+    entry = kb.Entry(
+        title=VULNERABILITY_TITLE,
+        risk_rating="CRITICAL",
+        short_description=VULNERABILITY_DESCRIPTION,
+        description=VULNERABILITY_DESCRIPTION,
+        references={
+            "nvd.nist.gov": f"https://nvd.nist.gov/vuln/detail/{VULNERABILITY_REFERENCE}"
+        },
+        recommendation="Ensure OpenSSH is updated to the latest version.",
+        security_issue=True,
+        privacy_issue=False,
+        has_public_exploit=True,
+        targeted_by_malware=False,
+        targeted_by_ransomware=False,
+        targeted_by_nation_state=False,
+    )
+    technical_detail = f"Target: {target.origin}:{target.port}\nDetails: {details}"
+    vulnerability = definitions.Vulnerability(
+        entry=entry,
+        technical_detail=technical_detail,
+        risk_rating=agent_report_vulnerability_mixin.RiskRating.CRITICAL,
+    )
+    return vulnerability
+
+
+def get_ssh_banner(ip_address: str, port: int) -> str:
+    try:
+        with socket.create_connection(
+            (ip_address, port), timeout=DEFAULT_TIMEOUT
+        ) as sock:
+            sock.settimeout(2)
+            banner = sock.recv(1024).decode().strip()
+            return banner
+    except Exception as e:
```

Comment:
Do not catch generic exceptions, [be explicit](https://google.github.io/styleguide/pyguide.html#:~:text=Never%20use%20catch%2Dall%20except%3A%20statements%2C%20or%20catch%20Exception%20or%20StandardError%2C%20unless%20you%20are)